packages/medusa/src/api/admin/product-tags/validators.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from "zod"
+import {
+  createFindParams,
+  createOperatorMap,
+  createSelectParams,
+} from "../../utils/validators"
+
+export type AdminGetProductTagParamsType = z.infer<
+  typeof AdminGetProductTagParams
+>
+export const AdminGetProductTagParams = createSelectParams()
+
+export type AdminGetProductTagsParamsType = z.infer<
+  typeof AdminGetProductTagsParams
+>
+export const AdminGetProductTagsParams = createFindParams({
+  limit: 20,
+  offset: 0,
+}).merge(
+  z.object({
+    q: z.string().optional(),
+    id: z.union([z.string(), z.array(z.string())]).optional(),
+    value: z.union([z.string(), z.array(z.string())]).optional(),
+    created_at: createOperatorMap().optional(),
+    updated_at: createOperatorMap().optional(),
+    deleted_at: createOperatorMap().optional(),
+    $and: z.lazy(() => AdminGetProductTagsParams.array()).optional(),
+    $or: z.lazy(() => AdminGetProductTagsParams.array()).optional(),
+  })
+)
+
+export type AdminCreateProductTagType = z.infer<typeof AdminCreateProductTag>
+export const AdminCreateProductTag = z
+  .object({
+    value: z.string(),
+    metadata: z.record(z.string(), z.unknown()).optional(),
+  })
+  .strict()
```

Comment:
thought: Should we default to `strict()` validation? This has fooled me a few times.

Is there a use case for wanting to let unexpected props pass through?

Reply:
@olivermrbl Yes I think we can be strict by default. The only issue is that we can't apply it globally in an easy way, since if we use preprocess (eg. to parse the data) the return type is ZodEffect, which can't have `strict()` applied to it. 

Anyway, let me just open a ticket and we can see if we can find a way around it.